### ISSUE TITLE
Big overhead on task switch with CONFIG_VIRT_CPU_ACCOUNTING_NATIVE 

### ISSUE THREAD

**Issue by chleroy** [linuxppc/issues#171]
Measurement done on a 8xx with the following app:

```
void *thread(void *arg)
{
	int i;

	for (i = 0; i < atoi((char*)arg); i++)
		pthread_yield();
}

int main(int argc, char **argv)
{
	pthread_t th1, th2;

	pthread_create(&th1, NULL, thread, argv[1]);
	pthread_create(&th2, NULL, thread, argv[1]);
	pthread_join(th1, NULL);
	pthread_join(th2, NULL);

	return 0;
}

```

Result with CONFIG_TICK_CPU_ACCOUNTING:

 ```
Performance counter stats for 'chrt -f 98 ./sched 100000' (50 runs):

       6633.062518      task-clock (msec)         #    0.955 CPUs utilized            ( +-  0.21% )
            200004      context-switches          #    0.030 M/sec                    ( +-  0.00% )

```

Result with CONFIG_VIRT_CPU_ACCOUNTING_NATIVE:

```
 Performance counter stats for 'chrt -f 98 ./sched 100000' (50 runs):

       8228.476465      task-clock (msec)         #    0.954 CPUs utilized            ( +-  0.23% )
            200004      context-switches          #    0.024 M/sec                    ( +-  0.00% )

```

So 1595ms additional ms are spent with CONFIG_VIRT_CPU_ACCOUNTING_NATIVE, that is a 24% overhead.

**Comment by chleroy** [comment]
From @mpe:

> PURR is Power5/6/7/8/9 and PA6T (pasemi). SPURR is Power6/7/8/9.
> 
> So we could easily flip PPC64 for PPC_BOOK3S_64, which would mean 64-bit
> Book3E CPUs don't get that overhead.
> 
> Beyond that is not so simple. We probably don't need that selected for
> bare metal kernels (powernv). But in practice all the distros build a
> multi platform kernel with powernv+pseries anyway.
> 
> We could turn it off on G5s (PPC970), by making it depend on POWERNV ||
> PSERIES || PPC_PASEMI, but I'm not sure if it's worth the trouble.
>

**Comment by chleroy** [comment]
Reducing scope to PPC_BOOK3S_64, see patch https://patchwork.ozlabs.org/patch/1151716/

**Comment by chleroy** [comment]
Merged in v5.4, see https://github.com/linuxppc/linux/commit/b4645ffc49cfe34f67feda20c34bd7a859c78312
